From 3879e9f5fac22f182f989ec8ba80fd75ac5fb810 Mon Sep 17 00:00:00 2001
From: Ramesh Rangavittal <ramesh.rangavittal@infineon.com>
Date: Thu, 9 Jun 2022 11:10:09 -0500
Subject: [PATCH 126/137] brcmfmac: Fix incorrect WARN_ON causing set_pmk
 failure

When STA switches from WPA2-PSK to Enterprise-PEAP
authentication, use_fwsup is set to BRCMF_PROFILE_FWSUP_ROAM.
This value also needs to be included in the WARN_ON check
for the authentication to go through.

Signed-off-by: Ramesh Rangavittal ramesh.rangavittal@infineon.com
---
 drivers/net/wireless/broadcom/brcm80211/brcmfmac/cfg80211.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/net/wireless/broadcom/brcm80211/brcmfmac/cfg80211.c b/drivers/net/wireless/broadcom/brcm80211/brcmfmac/cfg80211.c
index 2326084965de..5f8a5bbbd376 100644
--- a/drivers/net/wireless/broadcom/brcm80211/brcmfmac/cfg80211.c
+++ b/drivers/net/wireless/broadcom/brcm80211/brcmfmac/cfg80211.c
@@ -6106,6 +6106,7 @@ static int brcmf_cfg80211_set_pmk(struct wiphy *wiphy, struct net_device *dev,
 	ifp = netdev_priv(dev);
 	drvr = ifp->drvr;
 	if (WARN_ON((ifp->vif->profile.use_fwsup != BRCMF_PROFILE_FWSUP_1X) &&
+			(ifp->vif->profile.use_fwsup != BRCMF_PROFILE_FWSUP_ROAM) &&
 			(ifp->vif->profile.is_ft != true) &&
 			(ifp->vif->profile.is_okc != true)))
 		return -EINVAL;
-- 
2.17.1

